internal/pkg/cli/interfaces.go

```diff
@@ -249,10 +249,36 @@ type wsPipelineManifestReader interface {
 	ReadPipelineManifest(path string) (*manifest.Pipeline, error)
 }
 
+type pathDisplayer interface {
+	// DisplayPath takes any path and returns it in a form ready to be displayed to
+	// the user on the command line.
+	//
+	// No guarantees are given on the stability of the path across runs, all that is
+	// guaranteed is that the displayed path is visually pleasing & meaningful for a
+	// user.
+	//
+	// This path should not be stored in configuration files or used in any way except
+	// for being displayed to the user.
+	DisplayPath(path string) (string, error)
+}
+
+type wsPathRelativizer interface {
+	// Rel returns the path relative to the workspace root.
+	// The input path is allowed to be either relative to the current
+	// working directory or absolute.
+	//
+	// This is useful for storing a file path in configuration.
+	//
+	// Prefer another function for displaying a path to a user, as the
+	// workspace directory might be different from the current working
+	// directory.
```

Comment:
Similar to above, the interface can match other types besides a `*Workspace`, what do you think of just a single line:
```suggestion
	// Rel returns the path relative from the object to the target path.
```

Reply:
Makes sense. I think DisplayPath still deserves a sizable comment but since DisplayPath documents the necessary pitfalls the long comments are not needed.

However, since this interface's handling of relative paths is notably different than filepath.Rel, I'll leave that documented as a comment, if that is reasonable.